### PR TITLE
Corfu: evil-repeat support

### PR DESCRIPTION
**Pull request by aaronjensen**
Support for `evil-repeat` with Corfu.

See:

https://github.com/minad/corfu/pull/169
https://github.com/minad/corfu/pull/225

### CODE REVIEW

**Comment by condy0919** [comment]
```
modes/corfu/evil-collection-corfu.el:157:8: Error: the function ‘corfu-quit’ is not known to be defined.
```

**Comment by aaronjensen** [comment]
Yeah, that's odd. Any ideas?

**Comment by aaronjensen** [comment]
By the way, the Makefile doesn't actually work. It expects evil to be installed, but doesn't install it. Do you have instructions for running tests locally? That would allow me to iterate locally.

**Comment by aaronjensen** [comment]
There, that should work. The `declare-function` forms were incorrect.

**Comment by asymmetric** [comment]
The two listed corfu PRs were not merged, how does this affect whether this PR should be merged?

**Comment by aaronjensen** [comment]
It's the reason that I'm putting the code here. The author would prefer a different technique to support evil-repeat, but I do not know of any additional techniques. In other words, it not being merged there, is additional justification for it being merged here. I don't like having to patch one of corfu's methods, but there is no other option yet (aside from not supporting evil-repeat, of course).

File: modes/corfu/evil-collection-corfu.el [hunk 1]
```diff
@@ -89,7 +107,7 @@ This key theme variable may be refactored in the future so use with caution."
   ;; https://github.com/minad/corfu#tab-and-go-completion
   (when (memq 'tab-n-go evil-collection-corfu-key-themes)
     (setq corfu-cycle t
-          corfu-preselect-first nil)
+          corfu-preselect 'prompt)
```
Comment (aaronjensen) [inline modes/corfu/evil-collection-corfu.el:110]: This changed a bit ago, so `corfu-preselect-first` didn't actually exist anymore.

**Comment by aaronjensen** [comment]
By the way, if there's any other way to do this, either by modifying evil-repeat or another way to make corfu compatible, I'd be interested. I started a thread on Emacs-devel, but haven't heard anything yet.

**Comment by aaronjensen** [comment]
@condy0919 updated

**Pull request merged**